### ISSUE TITLE
setup for large genome assembly

### ISSUE THREAD

**Issue by ersgupta** [marbl/canu#848]
I am using canu 1.7 for assembling a genome of ~1.4gb size with ~70X pacbio data.
canu -p canu -d canu-pacbio useGrid=false maxThreads=60 maxMemory=500 genomeSize=1.4g -pacbio-raw file1 ....

I have the following queries:
1. What would be the expected runtime? I have few nodes with 1tb RAM and >100 cores (ofcourse there are other users also). For now I run on single node specifying the options above.
2. How much disk space can I expect it to occupy? 
3. I have not run this using the grid option, as I am not sure how much load this would put on the headnode, I use PBS. Any advise for this?
4. Are there any other recommended parameters which I should change for my setup?

Thanks :)

**Comment by ersgupta** [comment]
here is the configure output, if it helps.

```
> -- CONFIGURE CANU
> --
> -- Detected Java(TM) Runtime Environment '1.8.0_121' (from '...jdk1.8.0_121/bin/java').
> -- Detected gnuplot version '4.6 patchlevel 2' (from 'gnuplot') and image format 'png'.
> -- Detected 144 CPUs and 1008 gigabytes of memory.
> -- Limited to 500 gigabytes from maxMemory option.
> -- Limited to 60 CPUs from maxThreads option.
> -- No grid engine detected, grid disabled.
> --
> --                            (tag)Concurrency
> --                     (tag)Threads          |
> --            (tag)Memory         |          |
> --        (tag)         |         |          |     total usage     algorithm
> --        -------  ------  --------   --------  -----------------  -----------------------------
> -- Local: meryl    256 GB   32 CPUs x   1 job    256 GB   32 CPUs  (k-mer counting)
> -- Local: cormhap   32 GB   15 CPUs x   4 jobs   128 GB   60 CPUs  (overlap detection with mhap)
> -- Local: obtovl    16 GB   15 CPUs x   4 jobs    64 GB   60 CPUs  (overlap detection)
> -- Local: utgovl    16 GB   15 CPUs x   4 jobs    64 GB   60 CPUs  (overlap detection)
> -- Local: ovb        4 GB    1 CPU  x  60 jobs   240 GB   60 CPUs  (overlap store bucketizer)
> -- Local: ovs       32 GB    1 CPU  x  15 jobs   480 GB   15 CPUs  (overlap store sorting)
> -- Local: red        8 GB    4 CPUs x  15 jobs   120 GB   60 CPUs  (read error detection)
> -- Local: oea        4 GB    1 CPU  x  60 jobs   240 GB   60 CPUs  (overlap error adjustment)
> -- Local: bat      256 GB   16 CPUs x   1 job    256 GB   16 CPUs  (contig construction)
> -- Local: gfa       16 GB   16 CPUs x   1 job     16 GB   16 CPUs  (GFA alignment and processing)
> 
```

**Comment by skoren** [comment]
1/2. The runtime/space will vary with repeat content somewhat. If this is a vertebrate (fish, bird) it will likely run 1.5-2.5 weeks and take <2tb of space. If it's a plant/insect that's harder to predict. Probably a similar runtime but likely more space due to the repeats. There are ways to clean up space earlier than default or restrict repeat output in case you run into space issues.

3. Having more cores will be beneficial, there will be no load on the head node. All the computation runs on the cluster, the head node just submits the job. We have PBS support but don't have access to a PBS cluster for testing and it seems to change syntax with every version so you may need to spend a bit of time to get Canu submitting jobs on and running on your cluster.

4. Defaults are a pretty good start, take a look at the FAQ if you know your sample has heterozygosity you could use the suggested parameters there.

**Comment by ersgupta** [comment]
Thanks. This helps. :)

1/2. I should have mentioned earlier, it is a plant genome. Space issue is something I believe I would run into at a later stage. So far so good. I am halfway through cormhap step and using ~700gb. 

3. Cool. Can I kill the current cormhap step and start with grid? will it resume from where it leaves? Also, I can help you with some testing if you would like. I got the syntax sorted for now. :)

4. Unfortunately, it is unknown. For now, I will go ahead with the defaults and see how it goes. On this note, I have one more question, say I want to try some different parameters to play around and then evaluate for best assembly. For this, I should be able to use the already corrected reads right?

**Comment by skoren** [comment]
3. I don't think you can easily switch in the middle of the computation because the script is generated to not use grid environment variables. If you're already halfway through, you can wait for after correction and start on the grid with corrected reads. The largest space usage is correction, so if you can run through that without issues, you won't run out of space later.

4. For most settings, yes you can start from trimmed reads, skipping both correction and trimming.

**Comment by ersgupta** [comment]
Cool. Thanks. :)

**Comment by ersgupta** [comment]
So far so good. After 4 days of runtime I have reached generateCorrectionLayouts which has been running for 11 hrs to produce 16gb of seqDB.v001.dat. Size of canu.ovlStore being 1.6tb. I read in some posts that this step takes long time because it is single-threaded. I have a walltime of ~6 days left, is this step expected to complete before that? Any way i could speed this up?

**Comment by ersgupta** [comment]
Also sharing some numbers from the run till now. Do these numbers sound reasonable? or should I change some parameters and re-run.

```
-- Meryl finished successfully.
...
--    33899161 (max occurrences)
-- 123148628820 (total mers, non-unique)
--  2140741643 (distinct mers, non-unique)
--     5545915 (unique mers)
-- For mhap overlapping, set repeat k-mer threshold to 1231541.
--
-- Found 123154174735 16-mers; 2146287558 distinct and 5545915 unique.  Largest count 33899161.
--
-- OVERLAPPER (mhap) (correction)
--
-- Set corMhapSensitivity=low based on read coverage of 88.
--
-- PARAMETERS: hashes=256, minMatches=3, threshold=0.8
--
...

-- Overlap store 'correction/canu.ovlStore' successfully constructed.
-- Found 82781813970 overlaps for 13369751 reads; 434233 reads have no overlaps.
--
--
-- Purged 17632.404 GB in 11154 overlap output files.
-- Set corMinCoverage=4 based on read coverage of 88.
-- Global filter scores will be estimated.
-- Computing correction layouts.
```

Any help would be great. Thanks :)

**Comment by brianwalenz** [comment]
It's hard to say anything intelligent about the mer counts during correction.  There's just too much noise in them to really be useful.  The max occurrence is similar to what we see in human - 60x of PacBio, max occurrence = 152 million.

I'm not overly worried about run time.  Human took about 2 hours to make 19 GB of corStore.  IIRC, the overlaps for this are about the same size.  The ones that have been taking a long time have had much larger ovlStores.

You can estimate how much progress it has made with
```
strings -n 4 seqDB.v001.dat | wc -l
```
There will be one 'TIGR' string for each read (excluding reads with no overlaps).

Long term, it will end up parallelized or threaded, but nothing I can do in the immediate future to make it faster.

**Comment by ersgupta** [comment]
Thanks Brian, I understand. By the time I was checking the progress that step got over (seqDB.v001.dat: 18gb) now it is at correctReads, which should be over within a day.
I will keep posting the stats for your reference.

Here are the stats before correctReads:
```
[CORRECTION/LAYOUT]
--                             original      original
--                            raw reads     raw reads
--   category                w/overlaps  w/o/overlaps
--   -------------------- ------------- -------------
--   Number of Reads           12596077       1207907
--   Number of Bases       115226352018    5237938395
--   Coverage                    82.305         3.741
--   Median                        7466          2469
--   Mean                          9147          4336
--   N50                          13542          9811
--   Minimum                       1000             0
--   Maximum                     137402        102372
--   
--                                        --------corrected---------  ----------rescued----------
--                             evidence                     expected                     expected
--   category                     reads            raw     corrected            raw     corrected
--   -------------------- -------------  ------------- -------------  ------------- -------------
--   Number of Reads           11688914        3386738       3386738         123381        123381
--   Number of Bases       115164451053    59240720705   56000001615      399605390     339150345
--   Coverage                    82.260         42.315        40.000          0.285         0.242
--   Median                        8393          16124         15261           2117          1988
--   Mean                          9852          17491         16535           3238          2748
--   N50                          13710          17810         16718           3834          2891
--   Minimum                       1000          10782         10781           1002          1001
--   Maximum                     137402         137402        128532          42922         10780
--   
--                        --------uncorrected--------
--                                           expected
--   category                       raw     corrected
--   -------------------- ------------- -------------
--   Number of Reads           10293865      10293865
--   Number of Bases        60823964318   28415916657
--   Coverage                    43.446        20.297
--   Median                        4907          1089
--   Mean                          5908          2760
--   N50                           8504          7160
--   Minimum                          0             0
--   Maximum                     102372         10781
--   
--   Maximum Memory          1948437008
...
-- Using overlaps no worse than 0.3 fraction error for correcting reads (from corErrorRate parameter).
```

**Comment by ersgupta** [comment]
So the correction step got over, but the coverage has reduced drastically (from 80X to 35X). Could this be due to the corMinCoverage being bit high (4 in this case) which could be due to the slightly low quality data? Will reducing the coverage help recover data? 

```
--
-- In gatekeeper store './canu.gkpStore':
--   Found 3387032 reads.
--   Found 49578875909 bases (35.41 times coverage).
--
--   Read length histogram (one '*' equals 5730.28 reads):
--        0    999  14200 **
--     1000   1999  69082 ************
--     2000   2999  18910 ***
--     3000   3999  19096 ***
--     4000   4999  21712 ***
--     5000   5999  25699 ****
--     6000   6999  31851 *****
--     7000   7999  36711 ******
--     8000   8999  45629 *******
--     9000   9999  75248 *************
--    10000  10999 274817 ***********************************************
--    11000  11999 401120 **********************************************************************
--    12000  12999 363945 ***************************************************************
--    13000  13999 323367 ********************************************************
--    14000  14999 282051 *************************************************
--    15000  15999 244532 ******************************************
--    16000  16999 209855 ************************************
--    17000  17999 177902 *******************************
--    18000  18999 147823 *************************
--    19000  19999 121950 *********************
--    20000  20999  99917 *****************
--    21000  21999  81307 **************
--    22000  22999  65587 ***********
--    23000  23999  52577 *********
--    24000  24999  41943 *******
--    25000  25999  32970 *****
--    26000  26999  25653 ****
--    27000  27999  20056 ***
--    28000  28999  15433 **
--    29000  29999  11784 **
--    30000  30999   8906 *
--    31000  31999   6748 *
--    32000  32999   5062 
--    33000  33999   3748 
--    34000  34999   2769 
--    35000  35999   2044 
--    36000  36999   1454 
--    37000  37999   1043 
--    38000  38999    781 
--    39000  39999    515 
--    40000  40999    372 
--    41000  41999    283 
--    42000  42999    186 
--    43000  43999    129 
--    44000  44999     87 
--    45000  45999     57 
--    46000  46999     34 
--    47000  47999     25 
--    48000  48999     22 
--    49000  49999     12 
--    50000  50999      6 
--    51000  51999      6 
--    52000  52999      4 
--    53000  53999      2 
--    54000  54999      4 
--    55000  55999      1 
--    56000  56999      0 
--    57000  57999      1 
--    58000  58999      0 
--    59000  59999      0 
--    60000  60999      2 
--    61000  61999      0 
--    62000  62999      0 
--    63000  63999      0 
--    64000  64999      2 
--
-- Purging correctReads output after loading into stores.
-- Purged 921 .cns outputs.
-- Purged 1842 .out job log outputs.
...
----------------------------------------
-- Meryl finished successfully.
--
--  22-mers                                                                                           Fraction
--    Occurrences   NumMers                                                                         Unique Total
--       1-     1 3029741782 *******************************************************************--> 0.6972 0.0612
--       2-     2 377380235 ********************************************************************** 0.7840 0.0764
--       3-     4 218657715 ****************************************                               0.8170 0.0851
--       5-     7 106123340 *******************                                                    0.8451 0.0959
--       8-    11  79173764 **************                                                         0.8638 0.1072
--      12-    16  84812013 ***************                                                        0.8810 0.1230
--      17-    22  84806976 ***************                                                        0.9001 0.1480
--      23-    29  83130468 ***************                                                        0.9188 0.1816
--      30-    37  94443564 *****************                                                      0.9378 0.2266
--      38-    46  80222058 **************                                                         0.9594 0.2917
--      47-    56  38713092 *******                                                                0.9767 0.3563
--      57-    67  15706287 **                                                                     0.9847 0.3926
--      68-    79  10100086 *                                                                      0.9881 0.4111
--      80-    92   7459481 *                                                                      0.9903 0.4257
--      93-   106   5429709 *                                                                      0.9920 0.4384
--     107-   121   4171160                                                                        0.9932 0.4491
--     122-   137   3322150                                                                        0.9941 0.4585
--     138-   154   2654216                                                                        0.9949 0.4671
--     155-   172   2153301                                                                        0.9955 0.4748
--     173-   191   1776184                                                                        0.9960 0.4819
--     192-   211   1486438                                                                        0.9964 0.4883
--     212-   232   1260928                                                                        0.9967 0.4943
--     233-   254   1091786                                                                        0.9970 0.5000
--     255-   277    948241                                                                        0.9972 0.5053
--     278-   301    821685                                                                        0.9975 0.5104
--     302-   326    717947                                                                        0.9977 0.5151
--     327-   352    627955                                                                        0.9978 0.5197
--     353-   379    554945                                                                        0.9980 0.5240
--     380-   407    494501                                                                        0.9981 0.5280
--     408-   436    444088                                                                        0.9982 0.5320
--     437-   466    399963                                                                        0.9983 0.5357
--     467-   497    360927                                                                        0.9984 0.5394
--     498-   529    326392                                                                        0.9985 0.5429
--     530-   562    297392                                                                        0.9985 0.5462
--     563-   596    273093                                                                        0.9986 0.5495
--     597-   631    249158                                                                        0.9987 0.5527
--     632-   667    230950                                                                        0.9987 0.5558
--     668-   704    215551                                                                        0.9988 0.5588
--     705-   742    200756                                                                        0.9988 0.5618
--     743-   781    186726                                                                        0.9989 0.5647
--     782-   821    176766                                                                        0.9989 0.5676
--
--     3822481 (max occurrences)
-- 46478007166 (total mers, non-unique)
--  1316084594 (distinct mers, non-unique)
--  3029741782 (unique mers)
----------------------------------------
...
-- Reset obtOvlMerThreshold from auto to 47748.
--
-- Found 49507748948 22-mers; 4345826376 distinct and 3029741782 unique.  Largest count 3822481.
--
-- OVERLAPPER (normal) (trimming) erate=0.045
--

```

**Comment by skoren** [comment]
That's expected behavior, Canu only tries to correct the longest 40x of data. You can see in your read length distribution, most reads < 10kb were removed as being too short and only the longest were corrected. 

The k-mer distribution does have a peak at 30-37x coverage but it also has a large set of erroneous (1-count) kmers which could be due to the lower quality data. May still assemble OK, you can feed the histogram in 0-mercounts to something like genomescope (or post it here) to estimate heterozygosity and error rate in the data and confirm genome size.

If you want to have more data, you could increase corOutCoverage but we've found that 30-40x is typically ideal for assembly, adding more data doesn't really improve it (since the longest reads are already selected) and increases runtime. If you want to try to improve the quality/accuracy of the corrected data, you could run more than one round of correction. You'd run ```canu -correct corOutCoverage=100 corMinCoverage=0``` and then use the output corrected reads as input to a Canu run as you did before. I would suggest waiting for this assembly to finish first though before trying that.

**Comment by ersgupta** [comment]
Ok, that makes sense. And as you pointed out, simple increases runtime without adding much.

Here is the Genomescope output:
```
GenomeScope version 1.0
k = 22
property                      min               max               
Heterozygosity                1.47014%          1.49314%          
Genome Haploid Length         618,885,920 bp    621,401,776 bp    
Genome Repeat Length          275,589,063 bp    276,709,370 bp    
Genome Unique Length          343,296,857 bp    344,692,406 bp    
Model Fit                     91.7172%          98.6279%          
Read Error Rate               1.0448%           1.0448%           
```
http://genomescope.org/analysis.php?code=ff4xfPCZRaPiDfmhJVeP

I would run another run of correction with the corrected data (after the current run is over). Right now its at overlapping, which seems to take a long time (85 / 323 processes got over after 2.5 days). Any way I could speed that up? For now I am running 24 processes in parallel (15 cores and 16gb ram per process).

**Comment by skoren** [comment]
That is a very heterozygous genome, you will likely get double the haploid genome size in the output. I am concerned the genome size estimate is so off from your expectation. Didn't you say originally this is a 1.4gb genome? How accurate is that estimate?

If it really is this heterozygous and 1.4g then you have much lower coverage that you think because you should just think of it as two genomes. The corrected reads <20x and your input is then only 40x. Depending on your assembly from this run, you may need to re-run from the start with the heterozygous parameters on the FAQ which keep more corrected reads and adjust unitigging. Maybe include ```mhapSensitivity=normal``` too.

**Comment by ersgupta** [comment]
Sorry but I am not too sure what should be a ideal percentage for a less heterozygous genome, should it be <0.5% (heterozygosity of the genome is unknown)? DNA was extracted from clonally propagated plants. Regarding genome size, yes 1.4gb is the reported estimated genome size. I was also bit surprised by the estimate by genomescope but then it is based on illumina and I am not sure how it takes the read length parameter, so I ignored the genome size it reports.

I will let the current assembly finish and then evaluate to re-run following the FAQ.

On another note, I have a falcon-unzip assembly already. Is there any way I could use that with canu to improve it?

**Comment by ersgupta** [comment]
I am at the unitigging step which takes quite a while (using 130 cores). In the meantime "canu.ovlStore" directory in trimming takes about 2Tb space. Is this directory required after trimming in over? Is it safe to delete this?

And after trimming, the coverage got reduced to 33X.

```
-- Overlap store 'trimming/canu.ovlStore' successfully constructed.
-- Found 108293920954 overlaps for 3369403 reads; 10434581 reads have no overlaps.
...
# trimReads
----------------------------------------
--  PARAMETERS:
--  ----------
--     1000    (reads trimmed below this many bases are deleted)
--   0.0450    (use overlaps at or below this fraction error)
--        1    (break region if overlap is less than this long, for 'largest covered' algorithm)
--        1    (break region if overlap coverage is less than this many read, for 'largest covered' algorithm)
--  
--  INPUT READS:
--  -----------
--  13803984 reads  49578875909 bases (reads processed)
--       0 reads            0 bases (reads not processed, previously deleted)
--       0 reads            0 bases (reads not processed, in a library where trimming isn't allowed)
--  
--  OUTPUT READS:
--  ------------
--  3158777 reads  44180705520 bases (trimmed reads output)
--  187344 reads   2401633176 bases (reads with no change, kept as is)
--  10434581 reads     41619482 bases (reads with no overlaps, deleted)
--   23282 reads     79994861 bases (reads with short trimmed length, deleted)
--  
--  TRIMMING DETAILS:
--  ----------------
--  2586415 reads   1614717571 bases (bases trimmed from the 5' end of a read)
--  2647831 reads   1260205299 bases (bases trimmed from the 3' end of a read)
...
#splitReads
--  PARAMETERS:
--  ----------
--     1000    (reads trimmed below this many bases are deleted)
--   0.0450    (use overlaps at or below this fraction error)
--  INPUT READS:
--  -----------
--  3346121 reads  49457261566 bases (reads processed)
--  10457863 reads    121614343 bases (reads not processed, previously deleted)
--       0 reads            0 bases (reads not processed, in a library where trimming isn't allowed)
--  
--  PROCESSED:
--  --------
--       0 reads            0 bases (no overlaps)
--      96 reads       979306 bases (no coverage after adjusting for trimming done already)
--       0 reads            0 bases (processed for chimera)
--       0 reads            0 bases (processed for spur)
--  3346025 reads  49456282260 bases (processed for subreads)
--  
--  READS WITH SIGNALS:
--  ------------------
--       0 reads            0 signals (number of 5' spur signal)
--       0 reads            0 signals (number of 3' spur signal)
--       0 reads            0 signals (number of chimera signal)
--   15687 reads        16324 signals (number of subread signal)
--  
--  SIGNALS:
--  -------
--       0 reads            0 bases (size of 5' spur signal)
--       0 reads            0 bases (size of 3' spur signal)
--       0 reads            0 bases (size of chimera signal)
--   16324 reads      7290129 bases (size of subread signal)
--  
--  TRIMMING:
--  --------
--    7668 reads     36589785 bases (trimmed from the 5' end of the read)
--    8114 reads     38008603 bases (trimmed from the 3' end of the read)
...
--   Found 3346086 reads.
--   Found 46507712344 bases (33.21 times coverage).
--
--   Read length histogram (one '*' equals 5470.85 reads):
--        0    999      0 
--     1000   1999  82329 ***************
--     2000   2999  33289 ******
--     3000   3999  33259 ******
--     4000   4999  36932 ******
--     5000   5999  44910 ********
--     6000   6999  54266 *********
--     7000   7999  63801 ***********
--     8000   8999  81885 **************
--     9000   9999 117338 *********************
--    10000  10999 298034 ******************************************************
--    11000  11999 382960 **********************************************************************
--    12000  12999 343603 **************************************************************
--    13000  13999 301308 *******************************************************
--    14000  14999 261006 ***********************************************
--    15000  15999 223596 ****************************************
--    16000  16999 189249 **********************************
--    17000  17999 158752 *****************************
--    18000  18999 130655 ***********************
--    19000  19999 106750 *******************
--    20000  20999  86495 ***************
--    21000  21999  69225 ************
--    22000  22999  55770 **********
--    23000  23999  44380 ********
--    24000  24999  34786 ******
--    25000  25999  27139 ****
--    26000  26999  20995 ***
--    27000  27999  16103 **
--    28000  28999  12288 **
--    29000  29999   9377 *
--    30000  30999   6852 *
--    31000  31999   5213 
--    32000  32999   3837 
--    33000  33999   2860 
--    34000  34999   2053 
--    35000  35999   1427 
--    36000  36999   1074 
--    37000  37999    712 
...
-- Meryl finished successfully.
--
--  22-mers                                                                                           Fraction
--    Occurrences   NumMers                                                                         Unique Total
--       1-     1 2396199691 *******************************************************************--> 0.6613 0.0516
--       2-     2 334047014 ********************************************************************** 0.7535 0.0660
--       3-     4 196656492 *****************************************                              0.7890 0.0743
--       5-     7  98268772 ********************                                                   0.8196 0.0848
--       8-    11  76685236 ****************                                                       0.8407 0.0961
--      12-    16  83447564 *****************                                                      0.8609 0.1125
--      17-    22  83713444 *****************                                                      0.8833 0.1387
--      23-    29  83412797 *****************                                                      0.9056 0.1741
--      30-    37  93612830 *******************                                                    0.9285 0.2224
--      38-    46  76586316 ****************                                                       0.9540 0.2908
--      47-    56  35744547 *******                                                                0.9737 0.3563
--      57-    67  14741410 ***                                                                    0.9826 0.3920
--      68-    79   9640201 **                                                                     0.9864 0.4105
--      80-    92   7080907 *                                                                      0.9890 0.4254
--      93-   106   5147084 *                                                                      0.9908 0.4382
--     107-   121   3967310                                                                        0.9922 0.4490
--     122-   137   3155611                                                                        0.9933 0.4586
--     138-   154   2517856                                                                        0.9942 0.4673
--     155-   172   2037266                                                                        0.9948 0.4751
--     173-   191   1684020                                                                        0.9954 0.4822
--     192-   211   1407131                                                                        0.9959 0.4887
--     212-   232   1197287                                                                        0.9962 0.4948
--     233-   254   1037071                                                                        0.9966 0.5005
--     255-   277    896258                                                                        0.9969 0.5059
--     278-   301    778472                                                                        0.9971 0.5110
--     302-   326    677699                                                                        0.9973 0.5158
--     327-   352    593633                                                                        0.9975 0.5204
--     353-   379    525980                                                                        0.9977 0.5247
--     380-   407    468586                                                                        0.9978 0.5288
--     408-   436    420150                                                                        0.9979 0.5328
--     437-   466    377580                                                                        0.9980 0.5366
--     467-   497    342518                                                                        0.9982 0.5403
--     498-   529    308695                                                                        0.9982 0.5438
--     530-   562    282724                                                                        0.9983 0.5472
--     563-   596    258258                                                                        0.9984 0.5505
--     597-   631    236518                                                                        0.9985 0.5537
--     632-   667    219600                                                                        0.9985 0.5569
--     668-   704    205340                                                                        0.9986 0.5599
--     705-   742    189451                                                                        0.9987 0.5630
--     743-   781    177559                                                                        0.9987 0.5659
--     782-   821    170610                                                                        0.9988 0.5688
--
--     3574269 (max occurrences)
-- 44041244847 (total mers, non-unique)
--  1227234316 (distinct mers, non-unique)
--  2396199691 (unique mers)

```

**Comment by ersgupta** [comment]
By the time this canu runs gets over (takes ages for unitigging to get over, 190/287 done in ~2 weeks), I started assemble with `overlapper=mhap` and using grid support. Ran the command using bash on head node. The head node commands exits almost immediately after submitting a single core job. Now in unitigging at mhap step, I see 9 jobs running in parallel with 16 threads each (from the mhap.sh script), but qstat only shows the same single core job and not these 9 jobs. The node at which it is running has 144 cores, so it is treating the 144 cores to be free, whereas they are already occupied by other submitted jobs.

Here is the command I used in headnode:
```
~/apps/canu-1.7/Linux-amd64/bin/canu -assemble -p canu.asm -d canu.asm-pacbio gridOptions=' -N pb -l walltime=2400:00:00' gridEngineMemoryOption='-l mem=MEMORY' gridEngineThreadsOption='-l nodes=1:ppn=THREADS' useGrid=true overlapper=mhap genomeSize=1.4g -pacbio-corrected canu.trimmedReads.fasta.gz
```

**Comment by skoren** [comment]
I would guess your jobs are spending lots of time in the queue. It looks like your request is set to 100 days (2400 hours) and I'm sure the jobs aren't taking that much time each. You'd probably get more of them scheduled faster if you reduce that closer to how long they actually take.

As for your second question, I don't see any error. It submitted the 9 jobs to the grid and itself to wait for their completion. If they're not in queue then they must have finished, successfully or not. You can check their outputs in 1-overlapper. As for "so it is treating the 144 cores to be free, whereas they are already occupied by jobs submitted by other users". Canu doesn't treat any machine as free/not free. All it sees is that exists under scheduler control and submits jobs to the scheduler. The scheduler decides when/where to  run a job, not Canu.

**Comment by ersgupta** [comment]
Shouldn't a job complete after it has done what it is supposed to? irrespective of the walltime specified.

Maybe I was not clear earlier, so canu did not submit any jobs (I could not see any jobs other than the single core job it submitted in starting), but I could see 9 *.WORKING files in 1-overlapper/results and I saw 16 threads in the mhap.sh script.

**Comment by skoren** [comment]
Yes but the scheduler won't schedule a job if it knows it has to commit the machine to it for 100 days as fast as a job which will only reserve the machine for 10 days or 1 day.

I don't see how that could happen in the code. Can you post the ```canu.out``` file and one of the ```1-overlapper/mhap.*.out``` files?

**Comment by ersgupta** [comment]
2. I re-ran mhap manually for each of the 87 jobs separately so I do not have the old mhap*out, but here is a snippet from precompute.*out if it helps. Also let me know if you would like the complete canu.out file.

```
Running job 1 based on command line options.
Dumping reads from 1 to 96000 (inclusive).

Starting mhap precompute.

Running with these settings:
--filter-threshold = 5.0E-6
--help = false
--max-shift = 0.2
--min-olap-length = 500
--min-store-length = 0
--no-rc = false
--no-self = false
--no-tf = false
--num-hashes = 128
--num-min-matches = 5
--num-threads = 16
--ordered-kmer-size = 18
--ordered-sketch-size = 1000
--repeat-idf-scale = 10.0
--repeat-weight = 0.9
--settings = 0
--store-full-id = true
--supress-noise = 0
--threshold = 0.955
--version = false
-f = ../../0-mercounts/canu.asm.ms16.frequentMers.ignore.gz
-h = false
-k = 16
-p = ./000001.input.fasta
-q = .
-s = 

Reading in filter file ../../0-mercounts/canu.asm.ms16.frequentMers.ignore.gz.
Read in values for repeat 1240 and 0
Warning, k-mer filter file has zero elements.
Initializing
Trying size 1240 and it was successfull
Initialized
Time (s) to read filter file: 0.6214594480000001
Read in k-mer filter for sizes: [16]
Processing FASTA files for binary compression...
Current # sequences loaded and processed from file: 5000...
...
Current # sequences loaded and processed from file: 190000...
Processed 192000 sequences (fwd and rev).
Read, hashed, and stored file ./000001.input.fasta to ./000001.input.dat.
Time (s): 1175.796625613
Total time (s): 1175.797270991
```

**Comment by skoren** [comment]
It sounds like some of your nodes aren't allowed to submit jobs/use the grid. That is not compatible with Canu. That makes me think maybe your full assembly is also running on a single node, that would explain why it took a while. Have you checked the logs from that run in canu.out? 

You should ask if your IT will let your nodes submit jobs. Otherwise you have to run with useGrid=remote and then manually execute the jobs Canu tells you to run, wait for them to complete, and re-launch Canu to get any parallelism from the grid.

**Comment by ersgupta** [comment]
Got it. I will check with my IT admin to sort this part out.

About the full assembly (default parameters), it is running without grid so this issue is not applicable there, I specified 130 cores, using which it is running 10 jobs in parallel. However, I am monitoring the 1-overlapper folder and have starting running the jobs in batches from the reverse direction to use the other nodes. Also I am able to see all the jobs properly in qstat so no issues here.

One question here, for this step (unitigging overlap) what speeds up the process? more cores or more memory?

**Comment by skoren** [comment]
More cores are pretty generally the best option to speed up both mhap and overlapping.

**Comment by ersgupta** [comment]
Cool. I should be able to complete this step in full assembly by weekend I believe.

**Comment by ersgupta** [comment]
Great. Thanks. I will update the results soon.

**Comment by skoren** [comment]
The stats look OK, middle missing is a bit higher than I'd like to see (almost 1% is a bit high). Otherwise ok, looks to be a repetitive genome as well.

**Comment by ersgupta** [comment]
Thanks. I was waiting for the assembly to finish to reply back.
I am also running another run with overlapper=mhap, to see if I get similar stats with higher speed.

Anything I could change for the next run? I am running BUSCO for this assembly, right now. Also, I have hi-C data, which I will run now to see what sort of improvement I can get with this assembly.

The genome assembly is ~1.4x larger than the estimated genome size. (Longest contig ~4.1mb). Probably due to some repeats and heterozygosity?

I will run canu again with mhapSensitivity=normal and increasing the coverage. Any other option that could help?

Here are the remaining stats:
```
-- Configure RED for 8gb memory.
--                   Batches of at most (unlimited) reads.
--                                      500000000 bases.
--                   Expecting evidence of at most 540145162 bases per iteration.
--
--           Total                                               Reads                 Olaps Evidence
--    Job   Memory      Read Range         Reads        Bases   Memory        Olaps   Memory   Memory  (Memory in MB)
--   ---- -------- ------------------- --------- ------------ -------- ------------ -------- --------
--      1  8192.09         1-138818        28908    401231728  4592.64     45543703   521.21  1030.25
--      2  8192.13    138819-294816        29747    400510878  4584.42     46265772   529.47  1030.25
--      3  8192.08    294817-420978        28888    398635590  4562.93     48138572   550.90  1030.25
--      4  8192.01    420979-554840        29183    396448771  4537.91     50318157   575.85  1030.25
--      5  8192.37    554841-677884        29137    398204467  4558.01     48594079   556.12  1030.25
--      6  8192.07    677885-790404        28433    395313369  4524.90     51461702   588.93  1030.25
--      7  8192.03    790405-912699        28865    393596827  4505.27     53172871   608.52  1030.25
--      8  8192.10    912700-1026888       28717    389417730  4457.44     57358800   656.42  1030.25
--      9  8192.15   1026889-1168336       29306    393616210  4505.50     53162868   608.40  1030.25
--     10  8192.01   1168337-1309465       29688    390382205  4468.50     56383474   645.26  1030.25
--     11  8192.24   1309466-1431623       29481    395684626  4529.18     51101800   584.81  1030.25
--     12  8192.10   1431624-1560800       28990    394166739  4511.79     52608966   602.06  1030.25
--     13  8192.01   1560801-1683692       28786    390985368  4475.38     55783414   638.39  1030.25
--     14  8192.18   1683693-1807377       28837    391697063  4483.52     55085983   630.41  1030.25
--     15  8192.11   1807378-1926329       28829    391746331  4484.09     55030890   629.78  1030.25
--     16  8192.06   1926330-2035851       28790    391237296  4478.26     55535728   635.56  1030.25
--     17  8192.19   2035852-2138965       28005    397375937  4548.49     49410417   565.46  1030.25
--     18  8192.04   2138966-2274950       29218    398371440  4559.92     48398514   553.88  1030.25
--     19  8192.02   2274951-2402743       29013    396216849  4535.26     50551829   578.52  1030.25
--     20  8192.10   2402744-2529866       29353    396440813  4537.83     50334205   576.03  1030.25
--     21  8192.12   2529867-2651159       29078    393577801  4505.06     53199090   608.82  1030.25
--     22  8192.13   2651160-2765334       28829    393452859  4503.62     53326370   610.27  1030.25
--     23  8192.10   2765335-2868908       28295    397032021  4544.56     49745828   569.30  1030.25
--     24  8192.03   2868909-3001804       28791    396689341  4540.66     50080651   573.13  1030.25
--     25  8192.03   3001805-3126972       28484    389892845  4462.87     56877986   650.92  1030.25
--     26  8192.11   3126973-3249019       28489    390398946  4468.66     56378896   645.21  1030.25
--     27  8192.08   3249020-3365949       28563    394108548  4511.11     52666635   602.72  1030.25
--     28  8192.01   3365950-3476023       28238    388657280  4448.72     58112967   665.05  1030.25
--     29  8192.05   3476024-3577886       28029    392249160  4489.82     54524567   623.98  1030.25
--     30  8192.17   3577887-3699222       28142    394861551  4519.72     51922969   594.21  1030.25
--     31  8192.18   3699223-3841202       29129    390642995  4471.47     56139253   642.46  1030.25
--     32  8192.11   3841203-3981336       29009    386678526  4426.10     60097922   687.77  1030.25
--     33  8192.08   3981337-4117000       29046    385094136  4407.97     61679476   705.87  1030.25
--     34  8192.10   4117001-4241297       28976    388193151  4443.43     58582481   670.42  1030.25
--     35  8192.05   4241298-4351313       28451    391022277  4475.79     55750800   638.02  1030.25
--     36  8192.06   4351314-4478355       28493    391876527  4485.57     54897036   628.25  1030.25
--     37  8192.15   4478356-4611716       28871    384078085  4396.33     62702463   717.57  1030.25
--     38  8192.03   4611717-4756295       29013    378462879  4332.08     68306663   781.71  1030.25
--     39  8192.10   4756296-4903855       29057    375988777  4303.76     70786908   810.09  1030.25
--     40  8192.15   4903856-5040417       28776    373241855  4272.32     73538739   841.58  1030.25
--     41  8192.12   5040418-5169370       28844    393258574  4501.39     53519003   612.48  1030.25
--     42  8192.21   5169371-5316485       29485    390529038  4470.18     56255377   643.79  1030.25
--     43  8192.11   5316486-5468900       29620    388723581  4449.52     58051565   664.35  1030.25
--     44  8192.26   5468901-5615466       29441    386640783  4425.68     60147750   688.34  1030.25
--     45  8192.13   5615467-5750759       29347    385384123  4411.29     61393131   702.59  1030.25
--     46  8192.10   5750760-5853129       27746    392936786  4497.68     53842364   616.18  1030.25
--     47  8192.35   5853130-5939877       26534    399096042  4568.13     47708348   545.98  1030.25
--     48  8192.04   5939878-6044070       27450    400869072  4588.45     45905493   525.35  1030.25
--     49  8192.01   6044071-6183916       28976    390101590  4465.27     56666126   648.49  1030.25
--     50  8192.08   6183917-6331065       29442    391577845  4482.18     55194827   631.65  1030.25
--     51  8192.11   6331066-6476489       29176    384943566  4406.25     61832413   707.62  1030.25
--     52  8192.14   6476490-6611988       29259    388770832  4450.05     58007655   663.84  1030.25
--     53  8192.48   6611989-6731947       29016    394250330  4512.75     52558420   601.48  1030.25
--     54  8192.24   6731948-6853321       28424    398567257  4562.13     48222609   551.86  1030.25
--     55  8192.06   6853322-7016898       29898    397415282  4549.00     49354434   564.82  1030.25
--     56  8192.22   7016899-7179558       30064    396702681  4540.85     50080874   573.13  1030.25
--     57  8192.12   7179559-7329454       29929    395182992  4523.45     51592080   590.42  1030.25
--     58  8192.14   7329455-7461199       29402    399027311  4567.43     47751052   546.47  1030.25
--     59  8192.12   7461200-7625859       29408    396994279  4544.16     49782081   569.71  1030.25
--     60  8192.14   7625860-7805350       30007    393788164  4507.49     52988044   606.40  1030.25
--     61  8192.06   7805351-7968676       29927    392959147  4498.00     53810695   615.81  1030.25
--     62  8192.04   7968677-8116625       29614    395265372  4524.39     51503063   589.41  1030.25
--     63  8192.14   8116626-8224509       27750    401952662  4600.86     44830206   513.04  1030.25
--     64  8192.15   8224510-8341508       28211    395434426  4526.28     51347763   587.63  1030.25
--     65  8192.08   8341509-8463014       28130    391826631  4484.99     54949658   628.85  1030.25
--     66  8192.50   8463015-8574519       28108    392430112  4491.89     54382605   622.36  1030.25
--     67  8192.15   8574520-8679240       27953    392410557  4491.66     54372266   622.24  1030.25
--     68  8192.05   8679241-8764985       27011    399703614  4575.09     47073454   538.71  1030.25
--     69  8192.07   8764986-8855222       27358    400250535  4581.36     46526930   532.46  1030.25
--     70  8192.31   8855223-8945451       27651    400650457  4585.95     46147633   528.12  1030.25
--     71  8192.06   8945452-9029349       27260    401492982  4595.58     45283770   518.23  1030.25
--     72  8192.16   9029350-9116121       27115    401778497  4598.84     45007703   515.07  1030.25
--     73  8192.03   9116122-9217624       27972    400368743  4582.74     46403585   531.05  1030.25
--     74  8192.12   9217625-9313763       27929    401319543  4593.62     45461214   520.26  1030.25
--     75  8192.12   9313764-9442651       28322    393217673  4500.91     53562010   612.97  1030.25
--     76  8192.06   9442652-9600314       29350    380697732  4357.66     66073326   756.15  1030.25
--     77  8192.07   9600315-9757938       29087    377326358  4319.07     69446709   794.75  1030.25
--     78  8192.13   9757939-9895825       29007    385563635  4413.34     61214255   700.54  1030.25
--     79  8192.30   9895826-10009272      27925    385918384  4417.36     60877781   696.69  1030.25
--     80  8192.02  10009273-10083035      25537    393878225  4508.38     52899749   605.39  1030.25
--     81  8192.39  10083036-10158781      25806    394041391  4510.26     52768401   603.89  1030.25
--     82  8192.05  10158782-10237509      26218    398011456  4555.70     48767661   558.10  1030.25
--     83  8192.05  10237510-10309549      26706    400584255  4585.16     46193510   528.64  1030.25
--     84  8192.10  10309550-10387066      26782    401390914  4594.40     45391010   519.46  1030.25
--     85  8192.17  10387067-10466545      26953    401685504  4597.77     45101800   516.15  1030.25
--     86  8192.09  10466546-10546276      27088    401765741  4598.70     45014584   515.15  1030.25
--     87  8192.27  10546277-10625700      27352    402334651  4605.22     44460851   508.81  1030.25
--     88  8192.10  10625701-10702129      26849    398933243  4566.27     47847975   547.58  1030.25
--     89  8192.13  10702130-10791983      27553    403233994  4615.51     43548662   498.37  1030.25
--     90  8192.02  10791984-10887740      27887    400846743  4588.20     45925184   525.57  1030.25
--     91  8192.08  10887741-10983567      28234    402771046  4610.24     44004655   503.59  1030.25
--     92  8192.10  10983568-11076647      28195    403136974  4614.42     43641010   499.43  1030.25
--     93  8192.03  11076648-11248521      29075    391916175  4486.04     54853360   627.75  1030.25
--     94  8192.05  11248522-11411416      29727    397089824  4545.27     49679089   568.53  1030.25
--     95  8192.04  11411417-11556006      29769    398482104  4561.20     48285959   552.59  1030.25
--     96  8192.04  11556007-11664604      28955    398944064  4566.46     47826489   547.33  1030.25
--     97  8192.03  11664605-11749467      26728    399214011  4569.48     47561562   544.30  1030.25
--     98  8192.18  11749468-11842620      27574    400557568  4584.89     46229024   529.05  1030.25
--     99  8192.08  11842621-11936457      27682    400759124  4587.20     46018311   526.64  1030.25
--    100  8192.18  11936458-12028385      27704    400210241  4580.91     46575992   533.02  1030.25
--    101  8192.02  12028386-12118511      27870    400063541  4579.24     46708197   534.53  1030.25
--    102  8192.14  12118512-12207462      27712    402170401  4603.35     44612037   510.54  1030.25
--    103  8192.01  12207463-12300252      27433    398501639  4561.35     48270747   552.41  1030.25
--    104  8192.11  12300253-12403733      28281    402284241  4604.67     44494069   509.19  1030.25
--    105  8192.03  12403734-12508893      28447    403382801  4617.25     43388181   496.54  1030.25
--    106  8192.14  12508894-12609886      28310    400059398  4579.21     46721371   534.68  1030.25
--    107  8192.02  12609887-12705957      28293    401433483  4594.93     45337462   518.85  1030.25
--    108  8192.07  12705958-12794563      27958    402138877  4602.99     44637353   510.83  1030.25
--    109  8192.05  12794564-12883740      27346    400169526  4580.44     46605937   533.36  1030.25
--    110  8192.02  12883741-12989748      28141    401115182  4591.28     45655966   522.49  1030.25
--    111  8192.02  12989749-13095437      28076    399018479  4567.29     47752407   546.48  1030.25
--    112  8192.08  13095438-13202625      28518    399689708  4574.98     47085611   538.85  1030.25
--    113  8192.14  13202626-13305294      28520    402307586  4604.94     44472917   508.95  1030.25
--    114  8192.14  13305295-13400994      28363    403153753  4614.62     43627592   499.28  1030.25
--    115  8192.01  13400995-13496964      27670    402152562  4603.14     44619040   510.62  1030.25
--    116  8192.17  13496965-13617309      28442    400818021  4587.89     45965136   526.03  1030.25
--    117  8192.13  13617310-13734688      28469    399309448  4570.63     47470242   543.25  1030.25
--    118  6469.47  13734689-13803984      18782    267349465  3060.16     28928329   331.06  1030.25
--   ---- -------- ------------------- --------- ------------ -------- ------------ -------- --------
--                                                46507712344            6061635148
...
-- Loading read lengths.
-- Loading number of overlaps per read.
--
-- Configure OEA for 4gb memory.
--                   Batches of at most (unlimited) reads.
--                                      300000000 bases.
--
  FAILED - configured 13803984 jobs, but 13803984 jobs process 1024 reads or less each.  Increasing memory from 4 GB to 5 GB.
--
-- Configure OEA for 5gb memory.
--                   Batches of at most (unlimited) reads.
--                                      300000000 bases.
--
  FAILED - configured 13803984 jobs, but 13803984 jobs process 1024 reads or less each.  Increasing memory from 5 GB to 6.25 GB.
--
-- Configure OEA for 6.25gb memory.
--                   Batches of at most (unlimited) reads.
--                                      300000000 bases.
--
--           Total                                               Reads                 Olaps  Adjusts
--    Job   Memory      Read Range         Reads        Bases   Memory        Olaps   Memory   Memory  (Memory in MB)
--   ---- -------- ------------------- --------- ------------ -------- ------------ -------- --------
--      1  6400.05         1-88246         19583    275311529   271.16     31043685   947.38  3133.51
--      2  6400.31     88247-195609        20017    269066201   265.21     31246988   953.58  3133.51
--      3  6400.03    195610-299959        20167    272409995   268.41     31133353   950.11  3133.51
--      4  6400.03    299960-379905        18772    259118325   255.69     31550162   962.83  3133.51
--      5  6400.78    379906-465285        18746    257431810   254.08     31627352   965.19  3133.51
--      6  6400.01    465286-550012        18315    248295847   245.35     31887894   973.14  3133.51
--      7  6400.02    550013-632709        19303    263342919   259.73     31417229   958.78  3133.51
--      8  6400.03    632710-706938        18367    253199681   250.03     31735267   968.48  3133.51
--      9  6400.05    706939-773516        17101    238144054   235.63     32207935   982.91  3133.51
--     10  6400.19    773517-851453        18002    246615002   243.74     31946859   974.94  3133.51
--     11  6400.05    851454-923098        17325    236228970   233.81     32267291   984.72  3133.51
--     12  6400.24    923099-988307        16193    219589243   217.91     32794623  1000.81  3133.51
--     13  6400.18    988308-1050987       16448    223712755   221.85     32663703   996.82  3133.51
--     14  6400.30   1050988-1144741       18301    244612550   241.84     32012477   976.94  3133.51
--     15  6400.02   1144742-1232462       17508    230434846   228.29     32447409   990.22  3133.51
--     16  6400.38   1232463-1310252       16892    222154186   220.38     32718381   998.49  3133.51
--     17  6400.24   1310253-1387437       17886    236830740   234.41     32254114   984.32  3133.51
--     18  6400.15   1387438-1465248       18793    257636697   254.28     31600129   964.36  3133.51
--     19  6400.23   1465249-1542308       17326    234803460   232.45     32317810   986.26  3133.51
--     20  6401.03   1542309-1617336       17540    238659870   236.14     32223207   983.37  3133.51
--     21  6400.02   1617337-1687164       16299    220998140   219.26     32743466   999.25  3133.51
--     22  6400.24   1687165-1761960       17358    236378892   233.96     32268883   984.77  3133.51
--     23  6400.02   1761961-1832525       16670    225122105   223.20     32614094   995.30  3133.51
--     24  6400.45   1832526-1904866       17553    239209910   236.66     32187169   982.27  3133.51
--     25  6401.72   1904867-1967177       15850    215104165   213.62     32983747  1006.58  3133.51
--     26  6400.17   1967178-2031982       17274    234693161   232.35     32319387   986.31  3133.51
--     27  6400.14   2031983-2099886       18153    255270184   252.00     31674566   966.63  3133.51
--     28  6400.01   2099887-2177041       18839    264916461   261.22     31367938   957.27  3133.51
--     29  6400.02   2177042-2263204       18545    252811791   249.67     31746897   968.84  3133.51
--     30  6400.01   2263205-2345923       18685    255420387   252.16     31664956   966.34  3133.51
--     31  6400.03   2345924-2425244       18070    245677732   242.85     31970883   975.67  3133.51
--     32  6400.00   2425245-2505056       18466    249273680   246.29     31857016   972.20  3133.51
--     33  6400.61   2505057-2579351       17533    237105105   234.66     32258105   984.44  3133.51
--     34  6400.89   2579352-2654286       18119    245516967   242.70     32003941   976.68  3133.51
--     35  6400.75   2654287-2723023       17280    234432725   232.10     32346439   987.13  3133.51
--     36  6400.05   2723024-2789541       17094    235312284   232.93     32296435   985.61  3133.51
--     37  6400.01   2789542-2855823       18623    261353551   257.81     31479507   960.68  3133.51
--     38  6400.19   2855824-2942540       19098    265440185   261.73     31357228   956.95  3133.51
--     39  6400.62   2942541-3018013       16707    230171628   228.02     32475942   991.09  3133.51
--     40  6400.23   3018014-3089749       16397    225242858   223.31     32617612   995.41  3133.51
--     41  6401.21   3089750-3163943       16942    232040745   229.81     32436814   989.89  3133.51
--     42  6400.00   3163944-3233927       16439    224848445   222.93     32622356   995.56  3133.51
--     43  6400.46   3233928-3305453       17271    237249907   234.79     32249098   984.16  3133.51
--     44  6400.00   3305454-3374689       17103    236685667   234.24     32251845   984.25  3133.51
--     45  6401.08   3374690-3439943       16499    227522133   225.49     32574187   994.09  3133.51
--     46  6400.00   3439944-3498907       15761    216794394   215.23     32874678  1003.26  3133.51
--     47  6400.44   3498908-3558872       16375    229857430   227.71     32480086   991.21  3133.51
--     48  6400.01   3558873-3615681       16534    234187384   231.84     32330573   986.65  3133.51
--     49  6400.19   3615682-3705090       18123    251472902   248.38     31794943   970.30  3133.51
--     50  6400.52   3705091-3787702       16949    227122763   225.12     32567845   993.89  3133.51
--     51  6400.01   3787703-3867670       16494    220081202   218.39     32771465  1000.11  3133.51
--     52  6400.01   3867671-3945082       15912    212695691   211.33     33002922  1007.17  3133.51
--     53  6400.64   3945083-4018606       15415    204048376   203.07     33294214  1016.06  3133.51
--     54  6400.33   4018607-4090967       15630    208108184   206.94     33157172  1011.88  3133.51
--     55  6400.24   4090968-4161811       15578    207582555   206.44     33170501  1012.28  3133.51
--     56  6400.22   4161812-4231478       16688    223386771   221.55     32674930   997.16  3133.51
--     57  6400.19   4231479-4298816       16744    227160236   225.15     32556001   993.53  3133.51
--     58  6400.02   4298817-4361776       17192    240968570   238.33     32118509   980.18  3133.51
--     59  6401.13   4361777-4437717       16693    231005161   228.81     32466716   990.81  3133.51
--     60  6400.91   4437718-4514513       16655    223822702   221.96     32683842   997.43  3133.51
--     61  6400.22   4514514-4583508       15021    198959531   198.20     33439867  1020.50  3133.51
--     62  6400.11   4583509-4655295       14802    194698175   194.13     33569687  1024.47  3133.51
--     63  6400.47   4655296-4724214       13875    180330909   180.40     34031335  1038.55  3133.51
--     64  6400.85   4724215-4796826       14137    182940803   182.90     33962159  1036.44  3133.51
--     65  6400.13   4796827-4870100       14346    186113470   185.93     33839131  1032.69  3133.51
--     66  6400.05   4870101-4935919       13456    173884386   174.24     34219547  1044.30  3133.51
--     67  6400.92   4935920-4998622       13198    170421737   170.93     34356431  1048.48  3133.51
--     68  6400.18   4998623-5062331       13806    181502762   181.52     33985375  1037.15  3133.51
--     69  6400.50   5062332-5139905       18057    249965433   246.94     31852048   972.05  3133.51
--     70  6400.06   5139906-5229275       17794    236637212   234.22     32254488   984.33  3133.51
--     71  6400.19   5229276-5313750       17003    224920564   223.02     32625819   995.66  3133.51
--     72  6400.95   5313751-5395636       15821    207365201   206.24     33200491  1013.20  3133.51
--     73  6400.00   5395637-5483183       17210    225590228   223.66     32598504   994.83  3133.51
--     74  6400.00   5483184-5564017       16154    212175035   210.84     33018776  1007.65  3133.51
--     75  6400.02   5564018-5644984       16640    219470478   217.81     32790677  1000.69  3133.51
--     76  6400.29   5644985-5712857       14768    192694180   192.22     33638266  1026.56  3133.51
--     77  6400.02   5712858-5786781       16766    222695105   220.89     32690051   997.62  3133.51
--     78  6400.00   5786782-5845328       16801    241834131   239.14     32091184   979.35  3133.51
--     79  6400.00   5845329-5898748       16904    256142189   252.79     31644040   965.70  3133.51
--     80  6400.47   5898749-5961556       18531    275618840   271.42     31049033   947.54  3133.51--     81  6400.08   5961557-6030941       18512    272298612   268.25     31140156   950.32  3133.51
--     82  6400.04   6030942-6113989       16959    228423812   226.36     32511336   992.17  3133.51
--     83  6400.05   6113990-6194593       16812    226219194   224.25     32580853   994.29  3133.51
--     84  6400.40   6194594-6280026       17245    229744119   227.63     32481678   991.26  3133.51
--     85  6400.16   6280027-6361510       15955    210708963   209.43     33069915  1009.21  3133.51
--     86  6400.35   6361511-6440541       15921    208911807   207.72     33132221  1011.12  3133.51
--     87  6400.88   6440542-6519039       16075    214381443   212.94     32978672  1006.43  3133.51
--     88  6400.27   6519040-6598103       17238    230007327   227.88     32469161   990.88  3133.51
--     89  6400.50   6598104-6668994       16543    220110887   218.42     32786724  1000.57  3133.51
--     90  6400.09   6668995-6741796       18525    255968389   252.68     31650778   965.91  3133.51
--     91  6400.29   6741797-6810720       17991    256149183   252.83     31652038   965.94  3133.51
--     92  6401.24   6810721-6920098       20013    267630300   263.84     31322388   955.88  3133.51
--     93  6400.00   6920099-7023330       18672    247384408   244.49     31915934   974.00  3133.51
--     94  6400.00   7023331-7129645       19564    257890918   254.54     31586610   963.95  3133.51
--     95  6400.55   7129646-7230314       19049    252619318   249.50     31769923   969.54  3133.51
--     96  6400.33   7230315-7319784       18164    239476939   236.94     32174237   981.88  3133.51
--     97  6400.61   7319785-7407336       18938    253949262   250.76     31730224   968.33  3133.51
--     98  6400.81   7407337-7485881       19224    268256568   264.42     31289709   954.89  3133.51
--     99  6400.16   7485882-7600484       18590    246266990   243.43     31955980   975.22  3133.51
--    100  6400.72   7600485-7718307       19120    251128461   248.08     31822097   971.13  3133.51
--    101  6400.83   7718308-7820354       17610    231555016   229.37     32438619   989.95  3133.51
--    102  6400.00   7820355-7918074       17705    231729632   229.53     32406066   988.95  3133.51
--    103  6400.01   7918075-8011793       17902    234464484   232.15     32320502   986.34  3133.51
--    104  6400.09   8011794-8108889       18943    253046595   249.90     31741526   968.67  3133.51
--    105  6400.73   8108890-8179431       19068    277446940   273.18     30999797   946.04  3133.51
--    106  6400.60   8179432-8255522       18650    266031939   262.28     31352677   956.81  3133.51
--    107  6401.32   8255523-8328228       17470    244691312   241.89     32044593   977.92  3133.51
--    108  6400.09   8328229-8403935       17391    243115133   240.38     32053457   978.19  3133.51
--    109  6400.85   8403936-8470413       15700    217552305   215.95     32879035  1003.39  3133.51
--    110  6400.61   8470414-8536516       16390    228113950   226.05     32540312   993.05  3133.51
--    111  6401.30   8536517-8602168       17216    240610212   237.99     32171661   981.80  3133.51
--    112  6400.15   8602169-8663366       16385    229586451   227.45     32479312   991.19  3133.51
--    113  6400.43   8663367-8725067       18052    263254847   259.61     31434575   959.31  3133.51
--    114  6400.65   8725068-8779548       17672    263784185   260.10     31425741   959.04  3133.51
--    115  6400.70   8779549-8840206       18429    269150131   265.24     31258786   953.94  3133.51
--    116  6400.22   8840207-8903434       19159    277115608   272.86     30993554   945.85  3133.51
--    117  6400.32   8903435-8962586       18349    267363058   263.54     31302164   955.27  3133.51
--    118  6400.22   8962587-9020895       18854    277401628   273.13     30984966   945.59  3133.51
--    119  6400.19   9020896-9075138       18110    271189369   267.18     31178844   951.50  3133.51
--    120  6401.40   9075139-9142549       19310    280627614   276.22     30922335   943.67  3133.51
--    121  6400.90   9142550-9209497       18430    262957096   259.34     31459059   960.05  3133.51
--    122  6400.23   9209498-9276453       19162    275486479   271.31     31044723   947.41  3133.51
--    123  6400.09   9276454-9340331       19394    279538119   275.18     30913311   943.40  3133.51
--    124  6400.00   9340332-9421899       16331    225213752   223.28     32611010   995.21  3133.51
--    125  6400.18   9421900-9504903       15245    197995440   197.29     33468676  1021.38  3133.51
--    126  6400.31   9504904-9582118       14418    186404402   186.21     33835725  1032.58  3133.51
--    127  6400.48   9582119-9665042       15036    194613974   194.06     33584194  1024.91  3133.51
--    128  6400.26   9665043-9738551       13593    176239598   176.49     34152640  1042.26  3133.51
--    129  6400.26   9738552-9817422       15918    210249662   209.00     33087466  1009.75  3133.51
--    130  6400.06   9817423-9888135       15174    202065583   201.17     33337591  1017.38  3133.51
--    131  6401.04   9888136-9958793       15790    211895063   210.56     33061736  1008.96  3133.51
--    132  6400.02   9958794-10012513      14868    213295798   211.87     32985514  1006.64  3133.51
--    133  6400.35  10012514-10057272      15607    241659960   238.94     32109142   979.89  3133.51
--    134  6400.59  10057273-10102709      15648    239083842   236.49     32197561   982.59  3133.51
--    135  6401.14  10102710-10149017      15719    239975292   237.34     32187594   982.29  3133.51
--    136  6400.13  10149018-10198326      16435    248932227   245.90     31873840   972.71  3133.51
--    137  6401.02  10198327-10249290      17317    260042214   256.52     31555164   962.99  3133.51
--    138  6400.00  10249291-10297719      18235    276399477   272.15     31009599   946.34  3133.51
--    139  6400.45  10297720-10347421      17473    262969937   259.32     31444772   959.62  3133.51
--    140  6400.51  10347422-10401542      18431    274707801   270.54     31078784   948.45  3133.51
--    141  6400.01  10401543-10457123      18880    281237623   276.79     30857987   941.71  3133.51
--    142  6401.13  10457124-10511651      18583    276435623   272.20     31045130   947.42  3133.51
--    143  6400.39  10511652-10569205      19609    287974060   283.23     30659329   935.65  3133.51
--    144  6400.66  10569206-10623126      18606    275302068   271.12     31065120   948.03  3133.51
--    145  6400.04  10623127-10671006      17066    253168685   249.96     31738023   968.57  3133.51
--    146  6400.02  10671007-10728487      19054    282492952   277.99     30818809   940.52  3133.51
--    147  6400.02  10728488-10792950      19481    284234337   279.66     30764114   938.85  3133.51
--    148  6400.21  10792951-10857553      18793    269841994   265.91     31220698   952.78  3133.51
--    149  6400.38  10857554-10923869      19316    277934622   273.65     30972862   945.22  3133.51
--    150  6400.04  10923870-10990345      19696    279969005   275.60     30897859   942.93  3133.51
--    151  6401.19  10990346-11057808      20163    288734710   283.97     30661218   935.71  3133.51
--    152  6400.00  11057809-11145965      17847    247871835   244.93     31901518   973.56  3133.51
--    153  6400.23  11145966-11249095      17339    230826126   228.66     32442072   990.05  3133.51
--    154  6401.31  11249096-11356259      18991    254013808   250.83     31751251   968.97  3133.51
--    155  6400.01  11356260-11452533      18628    248303480   245.37     31887307   973.12  3133.51
--    156  6400.87  11452534-11547501      19811    265997150   262.28     31361693   957.08  3133.51
--    157  6400.02  11547502-11625235      19315    261894962   258.35     31462231   960.15  3133.51
--    158  6400.01  11625236-11684026      18217    263344865   259.70     31417925   958.80  3133.51
--    159  6400.30  11684027-11741323      17686    262785564   259.15     31445480   959.64  3133.51
--    160  6400.43  11741324-11803755      18569    270225427   266.27     31216068   952.64  3133.51
--    161  6400.15  11803756-11866924      18593    269964376   266.03     31215163   952.61  3133.51
--    162  6400.35  11866925-11931562      19069    275637955   271.45     31044004   947.39  3133.51
--    163  6400.01  11931563-11991812      18083    262098710   258.51     31456754   959.98  3133.51
--    164  6400.87  11991813-12055369      19344    278670825   274.35     30966100   945.01  3133.51
--    165  6400.19  12055370-12114445      18371    263542717   259.89     31417500   958.79  3133.51
--    166  6400.41  12114446-12178641      19584    282510790   278.02     30830627   940.88  3133.51
--    167  6400.04  12178642-12233607      17881    263735207   260.06     31406890   958.46  3133.51
--    168  6400.02  12233608-12297402      18066    259379622   255.92     31542083   962.59  3133.51
--    169  6400.15  12297403-12367616      19352    276832330   272.60     30999666   946.03  3133.51
--    170  6401.42  12367617-12442277      20086    283487354   278.97     30832874   940.94  3133.51
--    171  6400.01  12442278-12517410      20472    290585418   285.75     30564265   932.75  3133.51
--    172  6401.04  12517411-12585150      18962    268016308   264.18     31304916   955.35  3133.51
--    173  6400.25  12585151-12650893      18765    266974132   263.18     31311766   955.56  3133.51
--    174  6401.05  12650894-12716580      19834    281708201   277.26     30876348   942.27  3133.51
--    175  6400.50  12716581-12779033      19495    280623513   276.22     30892636   942.77  3133.51
--    176  6400.19  12779034-12832962      18282    269793070   265.85     31222136   952.82  3133.51
--    177  6401.22  12832963-12900157      18461    264042075   260.37     31435304   959.33  3133.51
--    178  6400.00  12900158-12973748      19654    280677275   276.27     30874447   942.21  3133.51
--    179  6400.00  12973749-13044390      18815    268113319   264.27     31268043   954.22  3133.51
--    180  6400.08  13044391-13112491      18048    254494366   251.26     31696765   967.31  3133.51
--    181  6400.09  13112492-13184648      19110    268150539   264.31     31269532   954.27  3133.51
--    182  6400.09  13184649-13254093      18912    266425279   262.66     31323424   955.92  3133.51
--    183  6400.06  13254094-13327092      20773    292891710   287.96     30493421   930.59  3133.51
--    184  6400.11  13327093-13392702      19493    277184611   272.94     30987182   945.65  3133.51
--    185  6401.28  13392703-13455351      19630    286166984   281.51     30744994   938.26  3133.51
--    186  6400.90  13455352-13532536      19081    273132509   269.06     31140176   950.32  3133.51
--    187  6400.34  13532537-13614038      19232    270241060   266.31     31212023   952.52  3133.51
--    188  6400.38  13614039-13693310      18978    265621229   261.90     31357984   956.97  3133.51
--    189  6400.49  13693311-13769640      19341    272064325   268.05     31159844   950.92  3133.51
--    190  5780.38  13769641-13803984       9685    139622436   141.45     14988645   457.42  3133.51
--   ---- -------- ------------------- --------- ------------ -------- ------------ -------- --------
--                                                46507712344            6061635148
--
...
-- Unitigger finished successfully.
-- Found, in version 1, after unitig construction:
--   contigs:      18521 sequences, total length 1961109191 bp (including 3695 repeats of total length 115653645 bp).
--   bubbles:      0 sequences, total length 0 bp.
--   unassembled:  274950 sequences, total length 2853476366 bp.
--
-- Contig sizes based on genome size --
--            NG (bp)  LG (contigs)    sum (bp)
--         ----------  ------------  ----------
--     10     1263479            78   140154522
--     20      869261           213   280005764
--     30      660536           400   420390458
--     40      514012           642   560406863
--     50      396314           954   700044636
--     60      300769          1362   840017394
--     70      229940          1896   980054255
--     80      177514          2593  1120083243
--     90      136170          3494  1260049556
--    100      101754          4686  1400045773
--    110       71917          6321  1540008569
--    120       49278          8695  1680026340
--    130       33385         12160  1820010175
--    140        7564         18278  1960002950
--
...
----------------------------------------
-- Found, in version 2, after consensus generation:
--   contigs:      18521 sequences, total length 1962338174 bp (including 3695 repeats of total length 115813290 bp).
--   bubbles:      0 sequences, total length 0 bp.
--   unassembled:  274950 sequences, total length 2853470289 bp.
--
-- Contig sizes based on genome size --
--            NG (bp)  LG (contigs)    sum (bp)
--         ----------  ------------  ----------
--     10     1262025            78   140154584
--     20      870162           213   280096733
--     30      660588           400   420541211
--     40      514990           641   560118368
--     50      396734           954   700342850
--     60      301164          1361   840106537
--     70      230076          1894   980090307
--     80      178007          2590  1120140665
--     90      136302          3489  1260067142
--    100      102061          4678  1400082861
--    110       72368          6307  1540022998
--    120       49445          8671  1680019474
--    130       33534         12121  1820009248
--    140        9546         18134  1960008149
--
```

**Comment by skoren** [comment]
You could try the heterozygous parameters for assembly along with corMhapSensitivity=normal as well if you haven't already set those to keep more corrected reads. Also, increased error rate if this is Sequel data (see issue #892 for some discussion on setting params for a heterozygous genome).

I'm also curious to see the ```unitigging/4-*/*003*sizes``` and the ```unitigging/4-*/*001*thr000.num000.log``` files from this run.

**Comment by ersgupta** [comment]
Ok, I will look into #892 . Yes, the data is Sequel.

Here are the files:
```
cat *003*sizes *001*thr000.num000.log                                                                                                                               [1/27]
CONTIGS (11425 tigs) (1859791604 length) (162782 average) (1.33x coverage)
ng010   4129869   lg010       23   sum   141255954  (CONTIGS)
ng020   2527744   lg020       70   sum   281632986  (CONTIGS)
ng030   1660606   lg030      139   sum   421581847  (CONTIGS)
ng040   1157727   lg040      240   sum   560468628  (CONTIGS)
ng050    855010   lg050      382   sum   700706273  (CONTIGS)
ng060    628578   lg060      572   sum   840541709  (CONTIGS)
ng070    442262   lg070      837   sum   980402452  (CONTIGS)
ng080    293353   lg080     1226   sum  1120219069  (CONTIGS)
ng090    198589   lg090     1813   sum  1260194270  (CONTIGS)
ng100    136984   lg100     2663   sum  1400063958  (CONTIGS)
ng110     90512   lg110     3922   sum  1540043705  (CONTIGS)
ng120     53141   lg120     5946   sum  1680009184  (CONTIGS)
ng130     28032   lg130     9550   sum  1820020362  (CONTIGS)

INITIAL EDGES
-------- ----------------------------------------
 2835960 reads are contained
10477250 reads have no best edges (singleton)
   27725 reads have only one best edge (spur) 
             4823 are mutual best
  463049 reads have two best edges 
            32651 have one mutual best edge
           409315 have two mutual best edges


ERROR RATES (6199634 samples)
-----------
mean   0.02027318 stddev 0.00869072 -> 0.07241752 fraction error =   7.241752% error
median 0.01880000 mad    0.00550000 -> 0.06772580 fraction error =   6.772580% error


EDGE FILTERING
-------- ------------------------------------------
10522880 reads have a suspicious overlap pattern
       0 reads had edges filtered
                0 had one
                0 had two
   42863 reads have length incompatible edges
            29794 have one
            13069 have two

FINAL EDGES
-------- ----------------------------------------
 2835960 reads are contained
10499223 reads have no best edges (singleton)
   28362 reads have only one best edge (spur) 
             4387 are mutual best
  440439 reads have two best edges 
            19793 have one mutual best edge
           385059 have two mutual best edges
```

**Comment by skoren** [comment]
Yep definitely sequel and you definitely want the heterozygous parameters. You could re-run just unitigging with the options ```correctedErrorRate=0.105 "batOptions=-dg 3 -db 3 -dr 1 -ca 500 -cp 50"```. I expect that will improve your asm.

If you want to re-start from scratch, I'd start with ```corMhapSensitivity=normal corOutCoverage=200 correctedErrorRate=0.105 "batOptions=-dg 3 -db 3 -dr 1 -ca 500 -cp 50"```

**Comment by ersgupta** [comment]
By the way, I also have a Falcon-unzip assembly for this data. BUSCO stats are slightly better than the falcon assembly, however the assembly size in falcon is almost equal to estimated size. In canu, you can see the duplicated %age is higher.

BUSCO stats:
Canu contigs-
Embryophyta: C:90.4%[S:42.1%,D:48.3%],F:2.2%,M:7.4%,n:1440
Eukaryota: C:88.1%[S:30.0%,D:58.1%],F:1.3%,M:10.6%,n:303

Falcon-unzip-
Embryophyta: C:87.3%[S:57.6%,D:29.7%],F:2.9%,M:9.8%,n:1440
Eukaryota: C:86.5%[S:44.6%,D:41.9%],F:1.7%,M:11.8%,n:303

**Comment by ersgupta** [comment]
Definitely sequel - because of the high error rate?
I will try the other options.

**Comment by skoren** [comment]
Definitely sequel because of the mean/median Canu reported:
```
median 0.01880000 mad    0.00550000 -> 0.06772580 fraction error =   6.772580% error
```

The default threshold is 4.5% so lower than the ideal on this dataset. RSII data typically has a median of 0.5% and here you're at 2% (which also is coming from the heterozygosity and from Sequel error).

Falcon-unzip runs Arrow as part of the final step so it gets polished with the raw data. You can run Arrow on the canu assembly to improve the busco stats further.

Falcon-unzip also tries to split the contigs into primary and alts but with the level of heterozygosity I think you have in your genome, there is going to be duplication in the primary in regions where Falcon-unzip didn't realize two contigs are from the same loci due to their high divergence (this typically starts happening with Falcon-unzip in genomes >1% divergence). Either way, you'll have to deduplicate either the Canu assembly or the Falcon-unzip assembly yourself.

**Comment by ersgupta** [comment]
Ok, got it. Will run these. Also, will try racon to see if that does better than Arrow.

**Comment by skoren** [comment]
No need to run Racon, it's a consensus caller using the fastq called data and Canu already calls consensus on the data. Arrow uses the raw signal information and a model trained on the specific error rate of each chemistry so it's going to give you a quality boost.

**Comment by ersgupta** [comment]
Ah ok. (y)

**Comment by ersgupta** [comment]
One last query for now, shall I use ```overlapper=mhap```? it provides huge speed boost.

**Comment by skoren** [comment]
If you're after the most contiguous assembly I would stick with the default. We haven't made mhap the default because it doesn't always generate as contiguous an assembly as the slower overlapper. If you do use overlapper=mhap also include utgReAlign=true as well.

**Comment by ersgupta** [comment]
Ok, I remember reading about utgReAlign in another post of yours. (y)

**Comment by ersgupta** [comment]
Also, I can delete the files from previous run, right? except corrected, trimmed and final assembly files?

I have gkpStore and unitigging/*ovlStore taking ~300GB.

**Comment by skoren** [comment]
Yes, you can delete those.

**Comment by skoren** [comment]
Original question answered and this issue has drifted a bit. I'm going to close. If you have an issue with the new assembly, please open a new issue.
